airflow/contrib/hooks/spark_jdbc_script.py

```diff
@@ -0,0 +1,142 @@
+    if create_table_column_types:
+        writer = writer.option("createTableColumnTypes", create_table_column_types)
+
+    writer \
+        .save(mode=save_mode)
+
+
+def spark_read_from_jdbc(spark, url, user, password, metastore_table, jdbc_table, driver,
+                         save_mode, save_format, fetch_size, num_partitions,
+                         partition_column, lower_bound, upper_bound):
+    reader = spark.read
+
+    # first set common options
+    reader = set_common_options(reader, url, jdbc_table, user, password, driver)
+
+    # now set specific read options
+    if fetch_size:
+        reader = reader.option('fetchsize', fetch_size)
+    if num_partitions:
+        reader = reader.option('numPartitions', num_partitions)
+    if partition_column and lower_bound and upper_bound:
+        reader = reader \
+            .option('partitionColumn', partition_column) \
+            .option('lowerBound', lower_bound) \
+            .option('upperBound', upper_bound)
+
+    reader \
+        .load() \
+        .write \
+        .saveAsTable(metastore_table, format=save_format, mode=save_mode)
+
+
+if __name__ == "__main__":  # pragma: no cover
+    # parse the parameters
+    parser = argparse.ArgumentParser(description='Spark-JDBC')
+    parser.add_argument('-cmdType', dest='cmd_type', action='store')
+    parser.add_argument('-url', dest='url', action='store')
+    parser.add_argument('-user', dest='user', action='store')
+    parser.add_argument('-password', dest='password', action='store')
+    parser.add_argument('-metastoreTable', dest='metastore_table', action='store')
+    parser.add_argument('-jdbcTable', dest='jdbc_table', action='store')
+    parser.add_argument('-jdbcDriver', dest='jdbc_driver', action='store')
+    parser.add_argument('-jdbcTruncate', dest='truncate', action='store')
+    parser.add_argument('-saveMode', dest='save_mode', action='store')
+    parser.add_argument('-saveFormat', dest='save_format', action='store')
+    parser.add_argument('-batchsize', dest='batch_size', action='store')
+    parser.add_argument('-fetchsize', dest='fetch_size', action='store')
+    parser.add_argument('-name', dest='name', action='store')
+    parser.add_argument('-numPartitions', dest='num_partitions', action='store')
+    parser.add_argument('-partitionColumn', dest='partition_column', action='store')
+    parser.add_argument('-lowerBound', dest='lower_bound', action='store')
+    parser.add_argument('-upperBound', dest='upper_bound', action='store')
+    parser.add_argument('-createTableColumnTypes',
+                        dest='create_table_column_types', action='store')
+    arguments = parser.parse_args()
+
+    # Disable dynamic allocation by default to allow num_executors to take effect.
+    spark = SparkSession.builder \
+        .appName(arguments.name) \
+        .enableHiveSupport() \
+        .getOrCreate()
+
+    if arguments.cmd_type == "import":
```

Comment:
Maybe change this. I guess this is legacy from sqoop. This part always made me cry since import and export does not really describe the direction of the data. Now we have the opportunity to fix this, my suggestion would be to change this to `spark_to_sql` and `sql_to_spark` to describe the direction of the flow 👍

Reply:
I've renamed them now to `spark_to_jdbc` and `jdbc_to_spark` to avoid any confusion with `spark-sql`

Reply:
Perfect, thanks! 👍